packages/react-core/src/components/JumpLinks/JumpLinks.tsx

```diff
@@ -100,6 +100,8 @@ export const JumpLinks: React.FunctionComponent<JumpLinksProps> = ({
   const [scrollItems, setScrollItems] = React.useState(hasScrollSpy ? getScrollItems(children, []) : []);
   const [activeIndex, setActiveIndex] = React.useState(activeIndexProp);
   const [isExpanded, setIsExpanded] = React.useState(isExpandedProp);
+  // Boolean to disable scroll listener from overriding active state of clicked jumplink
+  const isLinkClicked = React.useRef(false);
```

Comment:
Curious why you are using a ref here instead of state for a boolean value.

Reply:
Also curious about this

Reply:
@tlabaj @kmcfaul I did try using state with a boolean originally, but it wasn't working as expected - I'm not sure if either the `scrollSpy` was triggering before the state had finished updating, or if the state boolean was getting reset on rerender.

Either way, it does work as coded with `useRef` which [persists values across renders](https://reactjs.org/docs/hooks-reference.html#useref) or if I pull the boolean out of the component completely as a standalone variable, which I'm happy to switch to if that's preferred.

Reply:
After talking it through with Evan - this is a recommended pattern from the react docs https://reactjs.org/docs/hooks-reference.html#useref to store mutable instance fields (similar to how a private variable exists in a class). 

It can also work as a boolean stored outside the component. I'm good with either implementation